src/platform/forms-system/src/js/web-component-patterns/relationshipToVeteranPattern.jsx

```diff
@@ -106,6 +112,35 @@ export const claimantRelationshipToVeteranSpouseOrChildUI = personTitle => {
   };
 };
 
+/**
+ * Schema that takes a list of label keys, to be used in
+ * conjunction with customLabels property of relationshipToVeteranUI.
+ *
+ * This provides a way to have customized labels while still
+ * optionally taking advantage of the dropdown "other" behavior without
+ * reinventing the wheel.
+ *
+ * ```js
+ * relationshipToVeteran: customRelationshipSchema([
+ *  'spouse',
+ *  'caretaker',
+ *  'other',     // Include if you want the functionality
+ * ]),
+ * ```
+ * @param {array} labels array of strings that correspond to label keys
+ *   provided to relationshipToVeteranUI
+ * @returns {object}
+ */
+export const customRelationshipSchema = labels => {
```

Comment:
I think we should either do 
```
relationshipToVeteranUI({
   labels: {...}
})
+
relationshipToVeteranSchema(['label1', ...])
```
and convert `relationshipToVeteranSchema` to a function which accepts optional array, and update all the existing places

or add...

```
relationshipToVeteranCustomUI({
   labels: {...}
})
+
relationshipToVeteranCustomSchema(['label1', ...])
```

@ndsprinkle any thoughts on this?

Reply:
@rhasselle-oddball Thanks for tagging me on this! @michaelclement, what is the context of this change? Like I get that it's to add more flexibility to the pattern, but is that because the form you're working on has additional relationship to veteran options? Or is this just a general proposed change?

I ask that because most forms (10-10d as an example) have the relationship to veteran field as an empty text box, but some do have multiple options. If it's just an empty field, then I'd think we want a standardized option that isn't super customizable so that all the forms can share the same options for this recurring field. However, if the form **does** have specific options, then I'd lean more towards a "variant" of the main pattern that has the unique options hardcoded. This can then be reused if additional similar forms share those same options on their form.

And, having a totally customizable pattern just makes me feel like it's not really a pattern at all...

BUT all that being said, I'm hardly the police on all of this and I trust Robert's judgment more than my own, so I think my compromise would be going with the second option in Robert's comment. Creating a new, "variant" pattern that **only** accepts custom labels + supports the expanding "other" field in addition to a new schema that **only** accepts custom labels. That way the main pattern can stay relatively standardized while this new variant can be specifically for custom labels.

So, that's my recommendation. I'd prefer no customizable option and would instead go with a hardcoded variant, but if your team (and other teams as they're spun up) are going to be able to need to regularly customize the labels, then I'd go with the variant that supports customizable labels.

Reply:
I'm partial to the first option because we get to reuse more of the existing function - it looks like this pattern is only used a few times in the codebase so updating the schemas would be simple.

Reply:
_(Removed my first reply because Nick's comment hadn't been posted when I started typing)_ 
Thanks for the input! That makes a lot of sense - your point about a pattern that's totally customizable not really being a pattern is especially helpful when thinking about the utility of extending this vs just making something custom for 10-10d's use.

I think with all that under consideration, going the totally customizable route doesn't fit. And adding a new combination of relationships that only has hypothetical usage for others down the road doesn't really seem to warrant extending the form system here. 

If that seems fair, I'll just move the customized version down into 10-10d's code and close this effort.

Reply:
As a followup thought, would it make sense to have a general pattern that is a set of radio options with a dropdown for text entry under the last option. 

So rather than being focused on relationships, it's simply a way of saying "choose an option or enter one" without having to rebuild that mechanism from scratch. I realize this is basically still pure customization, but that does seem like an interaction pattern. Not sure if there's a place for something like that?